keel-core/src/main/kotlin/com/netflix/spinnaker/keel/pause/ActuationPauser.kt

```diff
@@ -66,44 +69,64 @@ class ActuationPauser(
   }
 
   fun resumeResource(id: String) {
     log.info("Resuming resource $id")
     pausedRepository.resumeResource(id)
     // helps a user not be confused by an out of date status from before a pause
-    publisher.publishEvent(ResourceActuationResumed(resourceRepository.get(id)))
+    publisher.publishEvent(ResourceActuationResumed(resourceRepository.get(id), clock))
   }
 
   fun pausedApplications(): List<String> =
     pausedRepository.getPausedApplications()
 
+  /**
+   * Adds a pause event to the resource history for every pause event from the parent application.
+   * We do this dynamically so that it applies to all resources in the app, even those added _after_ the
+   * application was paused.
+  */
   fun addSyntheticPausedEvents(originalEvents: List<ResourceEvent>, resource: Resource<*>) =
-    originalEvents.toMutableList().also { events ->
-      // For user clarity we add a pause event to the resource history for every pause event from the parent app.
-      // We do this dynamically here so that it applies to all resources in the app, even those added _after_ the
-      // application was paused.
-      val appPausedEvents = resourceRepository
-        .applicationEventHistory(resource.application, events.last().timestamp)
-        .filterIsInstance<ApplicationActuationPaused>()
+    originalEvents.toMutableList().let { events ->
+      val oldestResourceEvent = events.last()
+      val relevantAppEvents = resourceRepository
+        .applicationEventHistory(resource.application, oldestResourceEvent.timestamp)
+      val appPausedEvents = relevantAppEvents.filterIsInstance<ApplicationActuationPaused>()
+      val oldestAppPausedEvent = appPausedEvents.lastOrNull()
+      val oldestAppResumedEvent = relevantAppEvents.lastOrNull { it is ApplicationActuationResumed }
+
+      // Handles the case when a resource was added to the application *after* the application was paused, as follows:
+      // If the oldest resource event in history is ResourceCreated (i.e. the very first event for the resource)...
+      if (oldestResourceEvent is ResourceCreated &&
+        // ...and the oldest ApplicationActuationResumed in the life of the resource happened after the resource was created
+        oldestAppResumedEvent != null && oldestAppResumedEvent.timestamp.isAfter(oldestResourceEvent.timestamp) &&
+        // ...and either there's no ApplicationActuationPaused event or the oldest one is later the oldest resume (i.e. a different pause)
+        (oldestAppPausedEvent == null || oldestAppResumedEvent.timestamp.isBefore(oldestAppPausedEvent.timestamp))) {
+        // ...then add a synthetic ResouceActuationPaused event with the timestamp of resource creation
+        events.add(
+          events.indexOf(oldestResourceEvent),
+          ResourceActuationPaused(resource, "Resource actuation paused at the application level",
+            oldestResourceEvent.timestamp)
+        )
+      }
```

Comment:
This now handles the case where a resource is added to the application **after** the application is paused, which I missed in #842. It's is kinda of annoying to track that based on the application events as we can't just query the history of those events without limit, so I'm basing the query off of the oldest resource event, and inferring whether or not the app was paused before resource creation as explained in the comments.

Reply:
Well, that's kinda the point here... Since we didn't want to store `ResourceActuationPaused` events for every resource in the application when it is paused, we added this mechanism to inject fake ones when the API is called. Having said that, whether we should just inject the original `ApplicationActuationPaused` events in the history instead of changing the type, I'm not sure... I guess we could, but it would mix events with two different scopes (app and resource) in an endpoint that was intended to return only resource events.

@erikmunson any thoughts on this from the UI side?

Reply:
Here's a diagram to explain this algorithm if it helps...
![image](https://user-images.githubusercontent.com/1323478/76133774-ac7d0e80-5fce-11ea-9192-50e4f57d40b0.png)

Reply:
This diagram is really helpful.

I'm personally wary of adding the first _fake_ paused event because I worry that an operator will draw the wrong conclusion about when the pause was initiated.

I wonder if the UI itself could provide some sort if visual feedback that the app was paused before the resource was created without needing to create a synthetic event? Since we can assume that a resume without a matching pause implies that there was a pause that happened before the create, we could indicate this somehow in the UI view without creating a pause event that didn't actually happen at that time.

Reply:
That’s an excellent observation and question. I have to agree that if we could avoid that synthetic event with a weird timestamp, it would be best. On the other hand, the other alternative we had discussed was to actually emit such an event upon resource creation of the parent application was paused, which would effectively look exactly the same as depicted here.

@erikmunson, what are your thoughts?

Reply:
I've decided to move this to a separate PR (#853) to see if we can get the API changes needed for the UI merged sooner.